### PR TITLE
Exclude dependencies that come from Jenkins core

### PR DESCRIPTION
**Pull request by timja**
Fixes the upper bounds errors in:

https://github.com/jenkinsci/plugin-util-api-plugin/pull/57
https://github.com/jenkinsci/github-checks-plugin/pull/65
https://github.com/jenkinsci/checks-api-plugin/pull/37

cc @uhafner

### CODE REVIEW

**Review by XiongKezhi — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -48,6 +48,18 @@
       <groupId>edu.hm.hafner</groupId>
       <artifactId>codingstyle</artifactId>
       <version>${codingstyle.library.version}</version>
+      <exclusions>
+        <!-- Remove once once on at least 2.249 baseline -->
```
Comment (uhafner) [inline pom.xml:52]:
```suggestion
        <!-- Remove once on at least 2.249 baseline -->
```

File: pom.xml [hunk 2]
```diff
@@ -48,6 +48,18 @@
...
+        <exclusion>
+          <groupId>com.github.spotbugs</groupId>
+          <artifactId>spotbugs-annotations</artifactId>
+        </exclusion>
+        <!-- Remove once once on at least the baseline after 2.249 -->
```
Comment (uhafner) [inline pom.xml:57]:
```suggestion
        <!-- Remove once on at least the baseline after 2.249 -->
```

**Review by uhafner — APPROVED** [review]

**Comment by uhafner** [comment]
In which Jenkins baseline are the non-deprecated SpotBugs annotations?

**Comment by timja** [comment]
> In which Jenkins baseline are the non-deprecated SpotBugs annotations?

the comments are in the pom file in this PR, 2.249.x

**Comment by timja** [comment]
release notes are missing for this version btw

**Comment by uhafner** [comment]
Ah, thanks. Added them now: https://github.com/jenkinsci/plugin-util-api-plugin/releases/tag/v1.4.0

**Pull request merged**